staging/src/k8s.io/apiserver/pkg/endpoints/filters/trace.go

```diff
@@ -56,8 +57,16 @@ func WithTrace(handler http.Handler, longRunningCheck genericapirequest.LongRunn
 			utiltrace.Field{Key: "user-agent", Value: &internal.LazyTruncatedUserAgent{req}},
 			utiltrace.Field{Key: "client", Value: &internal.LazyClientIP{req}})
 		req = req.Clone(ctx)
-		// Set trace as root trace in context for nested tracing
-		defer trace.LogIfLong(30 * time.Second)
+		defer func(start time.Time) {
+			if time.Since(start) >= 30*time.Second {
+				// To mitigate https://github.com/kubernetes/kubernetes/issues/93042
+				// don't call LogIfLong here unless threshold is exceeded. This avoid printing
+				// this trace unconditionally when verbosity is set above 4, which is the case for
+				// e2e tests. LogIfLong called here instead of Log so that individual steps are
+				// only logged if step threshold is exceeded.
+				trace.LogIfLong(30 * time.Second)
```

Comment:
30s is super long, is this really the minimum? I think the best is to try to log the longest 1% - .1% of calls. That way it's a constant amount of logging and it's the most useful calls to log. Could do this by looking at the latency metric and logging if we're above 99th percentile, or something.

@wojtek-t might have thoughts.

Reply:
(I'm worried that this will make the trace essentially never show, which also isn't good?)

Reply:
the 30 second threshhold is preexisting IIUC

what is really unexpected to me is that LogIfLong **ignores** the specified duration at high verbosity levels

Reply:
> what is really unexpected to me is that LogIfLong ignores the specified duration at high verbosity levels

it looks like prior to https://github.com/kubernetes/kubernetes/commit/52838be7b22c53050d4e397c221f0b3c3842122c#diff-06ce45a2e81bbfd5d4ce7e3c67d49a84L120 that was not the case ... that seems like it should be reverted

Reply:
https://github.com/kubernetes/utils/pull/139/files#r456819382 was the location of the change in k8s.io/utils

Reply:
I agree with Jordan - the LogIfLong shouldn't ignore the passed threshold. And that is the thing that we should fix instead.

Re logging X% of calls - I'm not convinced. If everything is healthy we probably don't want to trace anything and that may result in significant increase in log volume in large clusters. We may get back to this discussion, but I think that we should first fix the regression Jordan pointed out above.

Reply:
> I agree with Jordan - the LogIfLong shouldn't ignore the passed threshold. And that is the thing that we should fix instead.

fixed in https://github.com/kubernetes/utils/pull/177 and pulled into k/k in https://github.com/kubernetes/kubernetes/pull/93253